lib/cli/src/commands/config.rs

```diff
@@ -37,13 +49,124 @@ pub struct Config {
     pkg_config: bool,
 }
 
+/// Subcommand for `wasmer config get | set`
+#[derive(Debug, Clone, PartialEq, Parser)]
+enum GetOrSet {
+    /// Get a value from the current wasmer config
+    #[clap(subcommand)]
+    Get(RetrievableConfigField),
+    /// Set a value in the current wasmer config
+    #[clap(subcommand)]
+    Set(StorableConfigField),
+}
+
+/// Value that can be queried from the wasmer config
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Parser)]
+pub enum RetrievableConfigField {
+    /// Print the path to the configuration file
+    #[clap(name = "config.path")]
+    ConfigPath,
+    /// Print the registry URL of the currently active registry
+    #[clap(name = "registry.url")]
+    RegistryUrl,
+    /// Print the token for the currently active registry or nothing if not logged in
+    #[clap(name = "registry.token")]
+    RegistryToken,
+    /// Print whether telemetry is currently enabled
+    #[clap(name = "telemetry.enabled")]
+    TelemetryEnabled,
+    /// Print whether update notifications are enabled
+    #[clap(name = "update-notifications.enabled")]
+    UpdateNotificationsEnabled,
+    /// Print the proxy URL
+    #[clap(name = "proxy.url")]
+    ProxyUrl,
+}
+
+/// Setting that can be stored in the wasmer config
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Parser)]
+pub enum StorableConfigField {
+    /// `registry.url`
+    #[clap(name = "registry.url")]
+    RegistryUrl(SetRegistryUrl),
+    /// `registry.token`
+    #[clap(name = "registry.token")]
+    RegistryToken(SetRegistryToken),
+    /// `telemetry.enabled`
+    #[clap(name = "telemetry.enabled")]
+    TelemetryEnabled(SetTelemetryEnabled),
+    /// `update-notifications.url`
+    #[clap(name = "update-notifications.enabled")]
+    UpdateNotificationsEnabled(SetUpdateNotificationsEnabled),
+    /// `proxy.url`
+    #[clap(name = "proxy.url")]
+    ProxyUrl(SetProxyUrl),
+}
+
+/// Set the current active registry URL
+#[derive(Debug, Clone, PartialEq, Eq, PartialOrd, Ord, Parser)]
+pub struct SetRegistryUrl {
```

Comment:
Where is this used? I don't see it anywhere?

Reply:
(same for the other setters)

Reply:
These were leftovers from when I was experimenting with clap on how to do CLI flags properly.

Reply:
Cool, then we shall remove them

Reply:
It's not a leftover, sorry, it's necessary so I can `#[derive(clap::Parser)]`

Reply:
We are using them actually, and they are needed. Resolving this comment